### PR TITLE
Use a KsqlSecurityExtension to register REST endpoints authorization

### PR DESCRIPTION
**Pull request by spena**
### Description 
This PR adds a `KsqlSecurityExtension` interface and a configuration that can be used to provide a security context to the `KsqlRestApplication`.

The configuration name is `ksql.security.extension.class` and accepts an implementation of `KsqlSecurityExtension`. This extension is initialized and REST endpoints registered when the `KsqlRestApplication.configureBaseApplication()` method is called when the server is started.

There is no default security for KSQL. Only a dummy implementation that is used internally to provide basic functionality to the `KsqlRestApplication`.

### Testing done 
Unit tests that verify `KsqlRestApplication` is initializing, registering, and closing the security extension.
I have my implementation that was called during my local functional tests.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 Confluent Inc.
```
Comment (vcrfxia) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java]: nit: year. Same comment for `KsqlSecurityExtension.java` below.

Comment (spena) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java]: Done

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * This is the default security extension for KSQL. For now, this class is just a dummy
+ * implementation of the {@link KsqlSecurityExtension} interface.
+ */
+public class KsqlDefaultSecurityExtension implements KsqlSecurityExtension {
+  @Override
+  public void initialize(final KsqlConfig config) throws KsqlException {
+  }
+
+  @Override
+  public void registerRestEndpoints(final Configurable<?> configurable) {
```
Comment (vcrfxia) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java]: Should this method throw a `KsqlException`? Same question for `close()` below.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import javax.ws.rs.core.Configurable;
+
+/**
+ * This is the default security extension for KSQL. For now, this class is just a dummy
+ * implementation of the {@link KsqlSecurityExtension} interface.
+ */
+public class KsqlDefaultSecurityExtension implements KsqlSecurityExtension {
+  @Override
+  public void initialize(final KsqlConfig config) throws KsqlException {
```
Comment (rodesai) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java]: KsqlException is a runtime exception - we don't really need to declare it.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java (hunk 2 above)
Comment (vcrfxia) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlDefaultSecurityExtension.java]: Ah, Rohan addressed my question above. Nvm.

**Pull request merged**